### PR TITLE
feat(fxgcppubsub): Added Avro and Protobuf schemas support

### PR DESCRIPTION
**Pull request by ekkinox**

### CODE REVIEW

File: fxgcppubsub/codec/codec.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+		schemaType:       schemaType,
+		schemaEncoding:   schemaEncoding,
+		schemaDefinition: schemaDefinition,
+	}
+}
+
+// Encode encodes an input into an avro or protobuf encoded slice of bytes.
+//
+//nolint:cyclop,exhaustive
+func (c *DefaultCodec) Encode(in any) ([]byte, error) {
```
Comment (Skyld) [inline fxgcppubsub/codec/codec.go:41]: How about splitting the codec per schema type, so if anyone wants to go with an alternative codec, they don't have to reimplement one for all schema types they don't necessarily use? Not sure if it worths the effort, just throwing in the idea

Comment (ekkinox) [reply fxgcppubsub/codec/codec.go:41]: high level publisher / subscriber completely obfuscate the fact there is auto ser/de, therefore that there is codecs involved behind.

imo overriding the codecs should be pretty rare, but if they do so, they just need to implement only the format support they need (raw, avro, proto)

File: fxgcppubsub/reactor/ack/reactor.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+// FuncNames returns the list of function names this reactor will react to.
+func (r *AckReactor) FuncNames() []string {
+	return []string{
+		"Acknowledge",
+	}
+}
+
+// React is the reactor logic.
+func (r *AckReactor) React(req interface{}) (handled bool, ret interface{}, err error) {
```
Comment (Skyld) [inline fxgcppubsub/reactor/ack/reactor.go:28]: error returned is always nil?

Comment (ekkinox) [reply fxgcppubsub/reactor/ack/reactor.go:28]: yes, it's a reactor chain, I just hook here to notify the waiters supervisor to stop, and let the chain continue, where the actual server will decide if there is an error or not

**Review by Skyld — APPROVED** [review]

File: fxgcppubsub/schema/normalize.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package schema
+
+import "strings"
+
+// NormalizeSchemaID normalizes a given schemaID
+// to be compatible with the "projects/{projectID}/schemas/{schemaID}" format.
+func NormalizeSchemaID(schemaID string) string {
+	if strings.Contains(schemaID, "/") {
```
Comment (ndk) [inline fxgcppubsub/schema/normalize.go:8]: Looks like `strings.LastIndex` :)

**Review by ndk — APPROVED** [review]

**Review by ndk — APPROVED** [review]

**Review by ndk — APPROVED** [review]

**Pull request merged**